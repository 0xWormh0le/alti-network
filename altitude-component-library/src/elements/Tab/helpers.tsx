export const getTabIndexOf = (element: HTMLElement) => {
  const tabs = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tab') as HTMLElement[]
  return tabs.indexOf(element)
}

export const getTabPanelIndexOf = (element: HTMLElement) => {
  const tabPanels = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tabPanel') as HTMLElement[]
  return tabPanels.indexOf(element)
}
