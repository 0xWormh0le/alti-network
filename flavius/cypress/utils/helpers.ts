export const capitalize = (input: string) => {
  if (input.length > 0) {
    const output = input.toLowerCase().split('')
    output[0] = output[0].toUpperCase()
    return output.join('')
  }
  return input
}
