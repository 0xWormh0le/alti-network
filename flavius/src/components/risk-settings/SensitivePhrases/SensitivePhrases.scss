@import 'styles/_shared.scss';

.SensitivePhrases {
  &__existing-phrases,
  .NewSensitivePhrase {
    padding: $SMALL_SPACING;
    border: 1px solid $PALE_GREY;
  }
}
