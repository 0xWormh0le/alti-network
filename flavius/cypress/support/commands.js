import * as Sentry from '@sentry/browser'
import { RISK_TYPE_IDS } from '../utils/config'
const { MANY_DOWNLOADS_BY_APP_ID } = RISK_TYPE_IDS
import { version } from '../../package.json'

Cypress.on('fail', (error, runnable) => {
  console.info('we failed!!', error, runnable)
  const errorMessage = `Heartbeat Error detected in spec '${runnable.parent.title}': ${error.message}`
  Sentry.init({
    dsn: 'https://4345e62a3abc48049f9965ccc17ed310@sentry.io/1306610',
    environment: Cypress.env('REACT_APP_STAGE'),
    release: `flavius@${version}`,
    integrations: [
      new Sentry.Integrations.Breadcrumbs({
        console: Cypress.env('REACT_APP_STAGE') === 'production',
        message: errorMessage
      })
    ]
  })
  const err = new Error(errorMessage)
  Sentry.withScope((scope) => {
    scope.setExtra('stacktrace', error.stack)
    scope.setExtra('codeframe', error.codeFrame)
    Sentry.captureException(err)
  })
  throw err
})

const checkMinimumValidPerson = (person) => {
  if (person) {
    expect(person.accessCount).to.be.a('number')
    expect(person.primaryEmail?.address).to.be.a('string')
    expect(person.hasOwnProperty('name')).to.be.true
  }
}

const checkMinimumFile = (file) => {
  const {
    createdAt,
    createdBy,
    externalAccessCount,
    externalAccessList,
    fileId,
    fileName,
    internalAccessCount,
    internalAccessList,
    lastModified,
    linkVisibility,
    lastIngested,
    iconUrl
  } = file
  if (createdAt) {
    expect(createdAt).to.be.a('number')
  }
  if (iconUrl) {
    expect(iconUrl).to.be.a('string')
  }
  expect(externalAccessCount).to.be.a('number')
  expect(externalAccessList).to.be.an('Array')
  // todo: uncomment. id and name is currently null
  expect(fileId).to.be.a('string')
  expect(fileName).to.be.a('string')
  expect(internalAccessCount).to.be.a('number')
  expect(internalAccessList).to.be.an('Array')
  if (lastModified) {
    expect(lastModified).to.be.a('number')
  }
  if (lastIngested) {
    expect(lastIngested).to.be.a('number')
  }
  expect(linkVisibility).to.be.a('string')
}

const checkMinimumFiles = (files) => checkMinimumFile(files[randomIndex(files)])

const randomIndex = (array) => Math.floor(Math.random() * array.length)

Cypress.Commands.add('login', () => {
  cy.get('#email').type('bobbie@thoughtlabs.io')
  cy.get('#password').type(Cypress.env('CYPRESS_USER_PASSWORD'))
  cy.get('.Login__button').click()
})

Cypress.Commands.add('checkMinimumValidPerson', checkMinimumValidPerson)

Cypress.Commands.add('checkFullPerson', (person) => {
  checkMinimumValidPerson(person)
  expect(person.externalCount).to.be.a('number')
  expect(person.userKind).to.be.a('number')
  expect(person.internalCount).to.be.a('number')
})

Cypress.Commands.add('checkFiles', (files) => {
  expect(files).to.be.an('Array')
  checkMinimumFiles(files)
})

Cypress.Commands.add('checkRisks', (risks, filteredRiskTypeId) => {
  expect(risks).to.be.an('Array')
  const randomRiskIndex = randomIndex(risks)
  const randomTestRisk = risks[randomRiskIndex]
  expect(randomTestRisk.platformId).to.be.a('string')
  expect(randomTestRisk.datetime).to.be.a('number')
  expect(randomTestRisk.fileCount).to.be.a('number')
  expect(randomTestRisk.hasOwnProperty('fileId')).to.be.true
  expect(randomTestRisk.hasOwnProperty('fileName')).to.be.true
  expect(randomTestRisk.riskDescription).to.be.a('string')
  expect(randomTestRisk.riskId).to.be.a('string')
  expect(randomTestRisk.riskTarget).to.be.an('Array')
  if (filteredRiskTypeId) {
    expect(randomTestRisk.riskTypeId).to.equal(filteredRiskTypeId)
  }
  expect(randomTestRisk.severity).to.be.a('number')

  checkMinimumValidPerson(randomTestRisk.creator)
  checkMinimumValidPerson(randomTestRisk.owner)

  if (filteredRiskTypeId === MANY_DOWNLOADS_BY_APP_ID) {
    expect(randomTestRisk.plugin.id).to.be.a('string')
    // todo: uncomment. name is currently null
    expect(randomTestRisk.plugin.name).to.be.a('string')
  }
})

Cypress.Commands.add('checkPagination', (paginationResponse, overrides, extra) => {
  expect(paginationResponse.orderBy).to.equal(
    overrides && overrides.hasOwnProperty('orderBy') ? overrides.orderBy : 'datetime'
  )
  expect(paginationResponse.pageCount).to.be.a('number')
  expect(paginationResponse.pageNumber).to.be.a('number')
  expect(paginationResponse.pageSize).to.be.a('number')
  expect(paginationResponse.sort).to.equal(overrides && overrides.hasOwnProperty('sort') ? overrides.sort : 'DESC')

  if (extra) {
    const { wrapper } = extra
    const { pageCount } = paginationResponse
    if (pageCount > 1) {
      if (pageCount > 5) {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', 5)
      } else {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', pageCount)
      }
    }
  }
})

Cypress.Commands.add('checkEvents', (events, filesIsObject) => {
  expect(events).to.be.an('Array')
  const randomEventIndex = randomIndex(events)
  const randomTestEvent = events[randomEventIndex]
  expect(randomTestEvent.datetime).to.be.a('number')
  expect(randomTestEvent.eventDescription).to.be.a('string')
  expect(randomTestEvent.eventId).to.be.a('string')
  expect(randomTestEvent.eventName).to.be.a('string')
  expect(randomTestEvent.targetPeople).to.be.an('Array')
  if (!filesIsObject) {
    expect(randomTestEvent.files).to.be.an('Array')
    const randomFileIndex = randomIndex(randomTestEvent.files)
    checkMinimumFile(randomTestEvent.files[randomFileIndex])
  } else {
    checkMinimumFiles(randomTestEvent.files)
  }
})
