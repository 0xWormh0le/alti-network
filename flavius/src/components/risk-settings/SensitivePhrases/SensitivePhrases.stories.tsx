import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import SensitivePhrases from './SensitivePhrases'

export default {
  title: 'RiskSettings/SensitivePhrases',
  component: SensitivePhrases
} as Meta

const Template: Story<any> = (args) => <SensitivePhrases {...args} />

export const Default = Template.bind({})
Default.args = {}
