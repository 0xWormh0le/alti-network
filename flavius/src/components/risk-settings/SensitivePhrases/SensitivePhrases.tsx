import React, { useCallback, useState, useRef, useEffect } from 'react'
import noop from 'lodash/noop'
import { ErrorBox } from '@altitudenetworks/component-library'
import EntityCountPagination from 'components/elements/EntityCountPagination'
import ModalConfirmationDialog from 'components/widgets/ModalConfirmationDialog'
import Phrase from 'components/elements/Phrase'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import useQueryParam from 'util/hooks/useQueryParam'
import UI_STRINGS from 'util/ui-strings'
import useSensitivePhraseApiClient from 'api/clients/sensitivePhraseApiClient'
import SubmittingModalContent from '../SubmittingModalContent'
import NewSensitivePhrase from '../NewSensitivePhrase'
import PhraseTable from '../PhraseTable'
import PhraseCount from '../PhraseCount'
import ConfirmPhraseModalContent from '../ConfirmPhraseModalContent'
import './SensitivePhrases.scss'

export const checkDuplicates = (sensitivePhrases: SensitivePhrase[], newPhrase: SensitivePhrase) =>
  Boolean(sensitivePhrases.find((item) => item.phrase === newPhrase.phrase))

export const SensitivePhrases: React.FC<{}> = () => {
  const { useGetSensitivePhrases } = useSensitivePhraseApiClient()

  const { useAddSensitivePhrase } = useSensitivePhraseApiClient({
    handleError: (error) => {
      const msg = error.response.data.message
      setErrorMessage(msg.slice(msg.indexOf(': ') + 2))
      setErrorModalType('adding')
    },
    handleSuccess: () => {
      if (resetForm.current) {
        resetForm.current()
      }
      getSensitivePhrases.call({ pageNumber })
      setPhraseAddedModalVisible(true)
    }
  })

  const { useDeleteSensitivePhrase } = useSensitivePhraseApiClient({
    handleError: (error) => {
      const msg = error.response.data.message
      setErrorMessage(msg.slice(msg.indexOf(': ') + 2))
      setErrorModalType('deleting')
    },
    handleSuccess: () => {
      if (sensitivePhrases && pageNumber > 1 && sensitivePhrases.length === 1) {
        setPageNumber(pageNumber - 1)
      } else {
        getSensitivePhrases.call({ pageNumber })
      }
      setPhraseDeletedModalVisible(true)
    }
  })

  const resetForm = useRef<() => void>()
  const [pageNumber, setPageNumber] = useQueryParam<number>('page', 1)
  const [sensitivePhrases, setSensitivePhrases] = useState<SensitivePhrase[]>([])
  const [confirmAddModalVisible, setConfirmAddModalVisible] = useState(false)
  const [confirmDeleteModalVisible, setConfirmDeleteModalVisible] = useState(false)
  const [selectedSensitivePhrase, setSelectedSensitivePhrase] = useState<SensitivePhrase | null>(null)
  const [phraseAddedModalVisible, setPhraseAddedModalVisible] = useState(false)
  const [phraseDeletedModalVisible, setPhraseDeletedModalVisible] = useState(false)
  const [errorModalType, setErrorModalType] = useState<'adding' | 'deleting' | null>(null)
  const [errorMessage, setErrorMessage] = useState<string | null>(null)

  const [response, sensitivePhrasesError, isLoading, getSensitivePhrases] = useGetSensitivePhrases(pageNumber)
  const [, , deletePending, deleteSensitivePhrase] = useDeleteSensitivePhrase()
  const [, , addPending, addSensitivePhrase] = useAddSensitivePhrase()

  useEffect(() => {
    setSensitivePhrases(response?.sensitivePhrases || [])
  }, [response])

  const handleAddClick = useCallback((newPhrase: SensitivePhrase, resetFormFn) => {
    if (newPhrase.phrase) {
      resetForm.current = resetFormFn
      setSelectedSensitivePhrase(newPhrase)
      setConfirmAddModalVisible(true)
    }
  }, [])

  const handleRemoveClick = useCallback((phrase: SensitivePhrase) => {
    setSelectedSensitivePhrase(phrase)
    setConfirmDeleteModalVisible(true)
  }, [])

  const handleAddSensitivePhrase = useCallback(() => {
    setConfirmAddModalVisible(false)
    if (!selectedSensitivePhrase) {
      return
    }
    addSensitivePhrase.call({
      phrase: selectedSensitivePhrase.phrase,
      exact: selectedSensitivePhrase.exact
    })
  }, [selectedSensitivePhrase, addSensitivePhrase])

  const handleDeleteSensitivePhrase = useCallback(() => {
    setConfirmDeleteModalVisible(false)
    if (!selectedSensitivePhrase) {
      return
    }
    if (selectedSensitivePhrase.id) {
      deleteSensitivePhrase.call({
        id: selectedSensitivePhrase.id
      })
    }
  }, [selectedSensitivePhrase, deleteSensitivePhrase])

  if (sensitivePhrasesError)
    return (
      <ErrorBox
        mainMessage={UI_STRINGS.ERROR_MESSAGES.SOMETHING_WRONG}
        secondaryMessage={UI_STRINGS.ERROR_MESSAGES.OUR_END}
      />
    )

  return (
    <div className='SensitivePhrases'>
      <NewSensitivePhrase onAdd={handleAddClick} />
      <div className='SensitivePhrases__existing-phrases'>
        <Typography variant={TypographyVariant.H3}>Active Phrases</Typography>
        <PhraseCount loading={isLoading} count={sensitivePhrases.length} />
        <PhraseTable
          loading={isLoading}
          sensitivePhrases={sensitivePhrases}
          hasError={sensitivePhrasesError || false}
          onDelete={handleRemoveClick}
        />
        <EntityCountPagination
          entityCount={response?.sensitivePhrases?.length || 0}
          pageNumber={response?.pageNumber || 0}
          pageSize={response?.pageSize || 0}
          pageCount={response?.pageCount || 0}
          onPageChange={setPageNumber}
        />
      </div>
      {confirmAddModalVisible && selectedSensitivePhrase && (
        <ModalConfirmationDialog
          onConfirm={handleAddSensitivePhrase}
          onCancel={() => setConfirmAddModalVisible(false)}
          verticalAlign='top'
          confirmButtonText={UI_STRINGS.BUTTON_LABELS.CORRECT}
          cancelButtonText={UI_STRINGS.BUTTON_LABELS.CANCEL}
          confirmButtonActionType='primary'
          cancelButtonActionType='secondary'
          dialogTitle={UI_STRINGS.SENSITIVE_PHRASES.CONFIRM_NEW_PHRASE}
          message={<ConfirmPhraseModalContent sensitivePhrase={selectedSensitivePhrase} action='add' />}
        />
      )}
      {confirmDeleteModalVisible && selectedSensitivePhrase && (
        <ModalConfirmationDialog
          onConfirm={handleDeleteSensitivePhrase}
          onCancel={() => setConfirmDeleteModalVisible(false)}
          verticalAlign='top'
          confirmButtonText={UI_STRINGS.SENSITIVE_PHRASES.YES_DELETE}
          cancelButtonText={UI_STRINGS.SENSITIVE_PHRASES.NO_DELETE}
          confirmButtonActionType='alert'
          cancelButtonActionType='secondary'
          dialogTitle={UI_STRINGS.SENSITIVE_PHRASES.CONFIRM_DELETION}
          message={<ConfirmPhraseModalContent sensitivePhrase={selectedSensitivePhrase} action='delete' />}
        />
      )}
      {phraseAddedModalVisible && selectedSensitivePhrase && (
        <ModalConfirmationDialog
          onConfirm={() => setPhraseAddedModalVisible(false)}
          onCancel={() => setPhraseAddedModalVisible(false)}
          verticalAlign='top'
          confirmButtonText={UI_STRINGS.BUTTON_LABELS.OK}
          confirmButtonActionType='primary'
          dialogTitle={UI_STRINGS.SENSITIVE_PHRASES.SENSITIVE_PHRASES_UPDATED}
          message={
            <Typography variant={TypographyVariant.H3}>
              <Phrase phrase={selectedSensitivePhrase.phrase} size='sm' /> {UI_STRINGS.SENSITIVE_PHRASES.HAS_BEEN_ADDED}
              .
            </Typography>
          }
        />
      )}
      {(addPending || deletePending) && (
        <ModalConfirmationDialog
          verticalAlign='top'
          dialogTitle={
            addPending ? UI_STRINGS.SENSITIVE_PHRASES.CONFIRM_NEW_PHRASE : UI_STRINGS.SENSITIVE_PHRASES.CONFIRM_DELETION
          }
          message={<SubmittingModalContent action={addPending ? 'add' : 'delete'} />}
          onConfirm={noop}
          onCancel={noop}
        />
      )}
      {phraseDeletedModalVisible && selectedSensitivePhrase && (
        <ModalConfirmationDialog
          onConfirm={() => setPhraseDeletedModalVisible(false)}
          onCancel={() => setPhraseDeletedModalVisible(false)}
          verticalAlign='top'
          confirmButtonText={UI_STRINGS.BUTTON_LABELS.OK}
          confirmButtonActionType='primary'
          dialogTitle={UI_STRINGS.SENSITIVE_PHRASES.SENSITIVE_PHRASES_UPDATED}
          message={
            <Typography variant={TypographyVariant.H3}>
              <Phrase phrase={selectedSensitivePhrase.phrase} size='sm' />{' '}
              {UI_STRINGS.SENSITIVE_PHRASES.HAS_BEEN_DELETED}.
            </Typography>
          }
        />
      )}
      {errorModalType && (
        <ModalConfirmationDialog
          onConfirm={() => setErrorModalType(null)}
          onCancel={() => setErrorModalType(null)}
          verticalAlign='top'
          confirmButtonText={UI_STRINGS.BUTTON_LABELS.OK}
          confirmButtonActionType='primary'
          dialogTitle={UI_STRINGS.SENSITIVE_PHRASES.ERROR}
          message={
            <Typography variant={TypographyVariant.H3}>
              {errorMessage ?? (
                <>
                  {UI_STRINGS.SENSITIVE_PHRASES.WE_ENCOUNTERED} {errorModalType}{' '}
                  {UI_STRINGS.SENSITIVE_PHRASES.THIS_PHRASE}.
                  <br />
                  {UI_STRINGS.SENSITIVE_PHRASES.PLEASE_TRY_AGAIN}.
                </>
              )}
            </Typography>
          }
        />
      )}
    </div>
  )
}

export default SensitivePhrases
