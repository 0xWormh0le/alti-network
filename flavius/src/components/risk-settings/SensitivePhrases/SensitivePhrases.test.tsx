import React from 'react'
import { waitForElement } from '@testing-library/react'
import API from '@aws-amplify/api/lib'

import { sensitivePhrasesResponseMock } from 'test/mocks'
import { SensitivePhrases, checkDuplicates } from './SensitivePhrases'
import { renderWithRouter } from 'test/support/helpers'

jest.mock('@aws-amplify/api/lib')

describe('SensitivePhrases', () => {
  beforeEach(() => {
    // Mocking server response, to return an empty array
    const apiGet: any = API.get
    apiGet.mockResolvedValue(sensitivePhrasesResponseMock)
  })

  it('renders correctly', async () => {
    const { container } = renderWithRouter(<SensitivePhrases />, {
      route: '/risk-settings',
      mountRoute: '/risk-settings'
    })
    await waitForElement(() => container.querySelectorAll('tr').length)
    expect(container).toMatchSnapshot()
  })
})

describe('checkDuplicates', () => {
  it('returns true if both phrase and exact fields are the same', () => {
    expect(
      checkDuplicates(sensitivePhrasesResponseMock.sensitivePhrases, sensitivePhrasesResponseMock.sensitivePhrases[0])
    ).toBe(true)
  })

  it('returns true if phrase fields match and exact fields do not match', () => {
    const sensitivePhrase = {
      phrase: sensitivePhrasesResponseMock.sensitivePhrases[0].phrase,
      exact: !sensitivePhrasesResponseMock.sensitivePhrases[0].exact
    }
    expect(checkDuplicates(sensitivePhrasesResponseMock.sensitivePhrases, sensitivePhrase)).toBe(true)
  })

  it('returns false if phrase fields do not match', () => {
    const sensitivePhrase1 = {
      phrase: 'test-not-match',
      exact: true
    }
    const sensitivePhrase2 = {
      phrase: 'test-not-match',
      exact: false
    }
    expect(checkDuplicates(sensitivePhrasesResponseMock.sensitivePhrases, sensitivePhrase1)).toBe(false)
    expect(checkDuplicates(sensitivePhrasesResponseMock.sensitivePhrases, sensitivePhrase2)).toBe(false)
  })
})
