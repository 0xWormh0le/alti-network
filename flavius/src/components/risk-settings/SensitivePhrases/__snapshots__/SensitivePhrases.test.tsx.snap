// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensitivePhrases renders correctly 1`] = `
<div>
  <div
    class="SensitivePhrases"
  >
    <form
      class="NewSensitivePhrase"
    >
      <h3
        class="Typography Typography--h3"
      >
        Add Sensitive Phrase
      </h3>
      <div
        class="NumberedList"
      >
        <div
          class="NumberedList__item"
        >
          <div
            class="NumberedList__col-number"
          >
            <div
              class="NumberedList__item-number"
            >
              1
            </div>
          </div>
          <div
            class="NewSensitivePhrase__field"
          >
            <input
              class="FormControl FormControl--flat"
              name="phrase"
              placeholder="Enter Sensitive Phrase"
              type="text"
            />
          </div>
        </div>
        <div
          class="NumberedList__item"
        >
          <div
            class="NumberedList__col-number"
          >
            <div
              class="NumberedList__item-number"
            >
              2
            </div>
          </div>
          <div
            class="NewSensitivePhrase__field"
          >
            <div
              class="NewSensitivePhrase__radios"
            >
              <div
                class="Radio Radio--inline NewSensitivePhrase__radio"
              >
                <label
                  class="Radio__label Radio__label--right"
                >
                  <input
                    checked=""
                    name="exact"
                    type="radio"
                    value="true"
                  />
                  <i
                    class="Radio__helper"
                  />
                  <span
                    class="Radio__label-text"
                  >
                    Exact Matching
                  </span>
                </label>
              </div>
              <div
                class="Radio Radio--inline NewSensitivePhrase__radio"
              >
                <label
                  class="Radio__label Radio__label--right"
                >
                  <input
                    name="exact"
                    type="radio"
                    value="false"
                  />
                  <i
                    class="Radio__helper"
                  />
                  <span
                    class="Radio__label-text"
                  >
                    Partial Matching
                  </span>
                </label>
              </div>
            </div>
            <div
              class="NewSensitivePhrase__help-box"
            >
              <div
                class="Typography Typography--body-tiny NewSensitivePhrase__helper-text"
              >
                For example, "int" would match the filename "Number or int" but would NOT match  "Internal Review".
              </div>
              <img
                alt="exact matching"
                class="NewSensitivePhrase__helper-img"
                src="exact-matching.svg"
              />
            </div>
          </div>
        </div>
        <div
          class="NumberedList__item"
        >
          <div
            class="NumberedList__col-number"
          >
            <div
              class="NumberedList__item-number"
            >
              3
            </div>
          </div>
          <button
            class="Button NewSensitivePhrase__button Button--primary Button--disabled"
            disabled=""
            type="submit"
          >
            Add
          </button>
        </div>
      </div>
    </form>
    <div
      class="SensitivePhrases__existing-phrases"
    >
      <h3
        class="Typography Typography--h3"
      >
        Active Phrases
      </h3>
      <p
        class="Typography Typography--body-small PhraseCount"
      >
        <span
          class="trigger"
        >
          5
           of 
          20
           available phrases
        </span>
      </p>
      <div
        class="Table PhraseTable"
      >
        <table
          class="Table__table"
        >
          <thead>
            <tr
              class="Table__heading"
            >
              <th
                class="Table__heading-cell"
              >
                <span>
                  sensitive phrase
                </span>
              </th>
              <th
                class="Table__heading-cell"
              >
                <span>
                  exact matching
                </span>
              </th>
              <th
                class="Table_heading-cell"
              />
            </tr>
          </thead>
          <tbody>
            <tr
              class="Table__row"
            >
              <td
                class="Table__cell"
              >
                <div
                  class="Phrase Phrase--sm"
                >
                  comp
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <svg
                  class="PhraseRow__icon"
                  height="9"
                  viewBox="0 0 13 9"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </td>
              <td
                class="Table__cell"
              >
                <button
                  class="Button PhraseTable__delete Button--pagination"
                  type="button"
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr
              class="Table__row"
            >
              <td
                class="Table__cell"
              >
                <div
                  class="Phrase Phrase--sm"
                >
                  financial
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <svg
                  class="PhraseRow__icon"
                  height="9"
                  viewBox="0 0 13 9"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </td>
              <td
                class="Table__cell"
              >
                <button
                  class="Button PhraseTable__delete Button--pagination"
                  type="button"
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr
              class="Table__row"
            >
              <td
                class="Table__cell"
              >
                <div
                  class="Phrase Phrase--sm"
                >
                  internal
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <svg
                  class="PhraseRow__icon"
                  height="9"
                  viewBox="0 0 13 9"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </td>
              <td
                class="Table__cell"
              >
                <button
                  class="Button PhraseTable__delete Button--pagination"
                  type="button"
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr
              class="Table__row"
            >
              <td
                class="Table__cell"
              >
                <div
                  class="Phrase Phrase--sm"
                >
                  earnings
                </div>
              </td>
              <td
                class="Table__cell"
              />
              <td
                class="Table__cell"
              >
                <button
                  class="Button PhraseTable__delete Button--pagination"
                  type="button"
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr
              class="Table__row"
            >
              <td
                class="Table__cell"
              >
                <div
                  class="Phrase Phrase--sm"
                >
                  external
                </div>
              </td>
              <td
                class="Table__cell"
              />
              <td
                class="Table__cell"
              >
                <button
                  class="Button PhraseTable__delete Button--pagination"
                  type="button"
                >
                  Delete
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="EntityCountPagination"
      >
        <div />
        <div
          class="EntityCountPagination__pagination-info"
        >
          <div
            class="EntityCountPagination__page-size EntityCountPagination__items-count"
          >
            <span>
              Displaying 0 Items Per Page
            </span>
          </div>
          <div
            class="Pagination"
          >
            <div
              class="Pagination__group"
            >
              <div
                class="PaginationItem PaginationItem--side"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  1
                </button>
              </div>
            </div>
            <div
              class="Pagination__group"
            >
              <div
                aria-label="prev"
                class="PaginationItem PaginationItem--arrow"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  <svg
                    class="Pagination__arrow"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polygon
                      fill="currentColor"
                      points="5,10 15,1 15,19"
                    />
                  </svg>
                </button>
              </div>
              <div
                aria-label="1"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  1
                </button>
              </div>
              <div
                aria-label="2"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  2
                </button>
              </div>
              <div
                aria-label="3"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  3
                </button>
              </div>
              <div
                aria-label="4"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  4
                </button>
              </div>
              <div
                aria-label="5"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  5
                </button>
              </div>
              <div
                aria-label="next"
                class="PaginationItem PaginationItem--arrow"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  <svg
                    class="Pagination__arrow"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polygon
                      fill="currentColor"
                      points="15,10 5,1 5,19"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="Pagination__group"
            >
              <div
                class="PaginationItem PaginationItem--side"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  10
                </button>
              </div>
            </div>
            <form
              class="PageNumberInput"
            >
              <div
                class="Typography Typography--body-tiny Typography--semibold PageNumberInput__label"
              >
                Go to page
              </div>
              <div
                class="PageNumberInput__input-wrapper"
              >
                <input
                  class="PageNumberInput__input"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="Button PageNumberInput__submit Button--pagination Button--disabled"
                disabled=""
                type="submit"
              >
                GO
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
